### DISCUSSION TITLE
Why is portals.conf not having any effect on ubuntu?

### DISCUSSION THREAD

**Discussion by dvhar** [flatpak/xdg-desktop-portal#1367] [Questions] [answered]
I wrote a new file chooser called [pikeru](https://github.com/dvhar/pikeru/tree/rust) and the configuration works perfectly on Arch/KDE, but I am unable to disable it via ~/.config/xdg-desktop-portal/portals.conf on ubuntu. The `UseIn` value in *.portal seems to be the only thing that determines which backend to use.
My test system (ubuntu 22.04) comes with no *portals.conf, so I provide my own when setting up my filepicker in that case. Here's what portals.conf looks like:
```
[preferred]
default=auto
org.freedesktop.impl.portal.FileChooser=pikeru
````
This is what /usr/share/xdg-desktop-portal/portals/pikeru.portal looks like:
```
[portal]
DBusName=org.freedesktop.impl.portal.desktop.pikeru
Interfaces=org.freedesktop.impl.portal.FileChooser;
UseIn=ubuntu
```
And this is what gtk.portal looks like for comparison:
```
[portal]
DBusName=org.freedesktop.impl.portal.desktop.gtk
Interfaces=org.freedesktop.impl.portal.FileChooser;(more interfaces...)
UseIn=gnome
```
XDG_CURRENT_DESKTOP is "ubuntu:GNOME". If I set UseIn=gnome in pikeru.portal, it will always use gtk. If I set UseIn=ubuntu, it will always use pikeru, unless I also set it to ubuntu in gtk.portal. My understanding is that UseIn indicates which DEs a backend can be used in and portals.conf has the highest precedence when telling it which to use, but it seems to be completely ignoring portals.conf. How can I make portals.conf determine which filepicker to use?
![Screenshot_20240531_064340](https://github.com/flatpak/xdg-desktop-portal/assets/33729230/c4f6829f-eb73-4690-b31b-42ffa3299fd5)

**Comment by dvhar** [comment, marked as answer]
Nevermind, tried it with a newer version of ubuntu and it works fine. I guess the configuration changed between xdg-desktop-portal versions 1.14 and 1.18

**Reply by dvhar** [reply]
I'd still be interested in how the older version gets configured, though it's not as urgent

**Reply by GeorgesStavracas** [reply]
The config file was introduced in xdg-desktop-portal 1.18, it didn't exist before. In the past it used to automatically pick portal backends using the session identifier.
